recipes/fast-cdr/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import textwrap	
+
+class FastCDRConan(ConanFile):
+
+    name = "fast-cdr"
+    license = "Apache-2.0"
+    homepage = "https://github.com/eProsima/Fast-CDR"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "eProsima FastCDR library for serialization"
+    topics = ("conan", "DDS", "Middleware", "Serialization")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared":          [True, False],
+        "fPIC":            [True, False]
+    }
+    default_options = {
+        "shared":            False,
+        "fPIC":              True
+    }
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+    _cmake = None
+
+    @property
+    def _pkg_cmake(self):
+        return os.path.join(
+            self.package_folder,
+            "lib",
+            "cmake"
+        )
+
+    @property
+    def _module_subfolder(self):
+        return os.path.join(
+            "lib",
+            "cmake"
+        )
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join(
+            self._module_subfolder,
+            "conan-target-properties.cmake"
+        )
+
+    @property
+    def _pkg_share(self):
+        return os.path.join(
+            self.package_folder,
+            "share"
+        )
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _create_cmake_module_alias_target(self):
+        content = ""
+        if self.options.shared and self.settings.os == "Windows":
+            # alias plus special properties which are required
+            content += textwrap.dedent("""\
+                if(TARGET fastcdr::fastcdr AND NOT TARGET fastcdr)
+                    add_library(fastcdr INTERFACE IMPORTED)
+                    set_property(TARGET fastcdr PROPERTY INTERFACE_LINK_LIBRARIES fastcdr::fastcdr)
+                    set_property(TARGET fastcdr PROPERTY INTERFACE_COMPILE_DEFINITIONS FASTCDR_DYN_LINK)
```

Comment:
This should be set in the `package_info` so it exposed by conan

Reply:
I apologize I was not very clear ... this line is particular needs to be set on the `self.cpp_info.defines` you can use `append(FASTCDR_DYN_LINK) as I recall

```suggestion
```

Once its removed the if and else are the same and could be simplified

Reply:
You my hero ;) i was searching exactly for sth like this and i could not find in conan - just in cmake. that's why i did. but cool. awesome. i bring this in.